manticore/core/executor.py

```diff
@@ -145,8 +207,24 @@ def locked_context(self):
                 visited.append(state.cpu.PC)
                 context['visited'] = visited
         '''
+        assert default in (list, dict, set)
         with self._lock:
-            yield self._shared_context
+            if key is None:
+                yield self._shared_context
+            elif '.' in key:
+                keys = key.split('.')
+                with self.locked_context('.'.join(keys[:-1])) as sub_context:
```

Comment:
I'm not sure string concatenation is the best way to handle the multiple key situation, can we try a list of keys?

Reply:
(Commenting for review) The concatenation is part of a shortcut syntax to use nested dicts. I.e. `locked_context('foo.bar')` will write to `_shared_context['foo']['bar']` and initialize `_shared_context['foo']` to `{}` if it doesn't exist.